pypeit/wavetilts.py

```diff
@@ -299,11 +318,138 @@ def trace_tilts(self, arcimg, lines_spec, lines_spat, thismask, slit_cen):
+            :attr:`msarc` with the model continuum.
+        """
+        # TODO: Instead check that extract arcs has been run using the
+        # "steps" attribute?
+        if self.arccen is None:
+            # Extract the arc spectra for all slits
+            self.arccen, self.arccen_bpm, self.arc_maskslit = self.extract_arcs()
+
+        # TODO: Should make this operation part of WaveTiltsPar ...
+        # Parse the upper and lower sigma rejection thresholds; used
+        # when rescaling continuum from center spectrum.
+        lower_rej, upper_rej = self.par['cont_rej'] if hasattr(self.par['cont_rej'], '__len__') \
+                                    else numpy.repeat(self.par['cont_rej'], 2)
+
+        # Fit the continuum of the extracted arc spectra for each slit
+        nspec, nslits = self.arccen.shape
+        spec = np.arange(nspec, dtype=float)
+        arc_continuum = np.zeros(self.arccen.shape, dtype=float)
+        arc_fitmask = np.zeros(self.arccen.shape, dtype=bool)
+        for i in range(nslits):
+            # TODO: What to do with the following iter_continuum parameters?:
+            #       sigthresh, sigrej, niter_cont, cont_samp, cont_frac_fwhm
+            arc_continuum[:,i], arc_fitmask[:,i] \
+                    = arc.iter_continuum(self.arccen[:,i], inmask=np.invert(self.arccen_bpm[:,i]),
+                                         fwhm=self.wavepar['fwhm'])
+            # TODO: Original version.  Please leave it for now.
+#            arc_fitmask[:,i], coeff \
+#                    = utils.robust_polyfit_djs(spec, self.arccen[:,i], self.par['cont_order'],
+#                                               function=self.par['cont_function'],
+#                                               minx=spec[0], maxx=spec[-1],
+#                                               upper=upper_rej, lower=lower_rej, use_mad=True,
+#                                               sticky=True)
+#            arc_continuum[:,i] = utils.func_val(coeff, spec, self.par['cont_function'],
+#                                                minx=spec[0], maxx=spec[-1])
+
+            if debug:
+                plt.plot(spec, self.arccen[:,i], color='k', label='Arc')
+                plt.scatter(spec, np.ma.MaskedArray(self.arccen[:,i], mask=arc_fitmask[:,i]),
+                            marker='x', s=10, color='C1', label='Ignored')
+                plt.plot(arc_continuum[:,i], color='C3', label='Cont. Fit')
+                plt.xlabel('Spectral pixel')
+                plt.ylabel('Counts')
+                plt.legend()
+                plt.show()
+
+        # For each slit, rescale the continuum to the spectrum at a
+        # given spatial position along the slit/order. This
+        # implementation may be too simplistic in how it treats the
+        # spatial axis.
+        nspat = self.slitmask.shape[1]
+        cont_image = np.zeros(self.msarc.image.shape, dtype=float)
+        ii = np.tile(np.arange(nspec), (nspat,1)).T
```

Comment:
Throughout the pipeline, I usually call these spec_img and spat_img which to me is more informative than ii and jj.

Reply:
Changed.